### PR TITLE
Update metric names to fit metrics naming conventions

### LINKED ISSUE #9714: Metrics names failed promlint linter
**What happened**:
We created a metric name linter based on promlint: https://github.com/kubevirt/kubevirt/pull/9822
This metrics names failed:
```
$ make lint-metrics
hack/dockerized "./hack/prom-metric-linter/metric_name_linter.sh"
go version go1.19.2 linux/amd64

go version go1.19.2 linux/amd64
kubevirt_migrate_vmi_pending_count: non-histogram and non-summary metrics should not have "_count" suffix
kubevirt_migrate_vmi_running_count: non-histogram and non-summary metrics should not have "_count" suffix
kubevirt_migrate_vmi_scheduling_count: non-histogram and non-summary metrics should not have "_count" suffix
kubevirt_vmi_cpu_affinity: counter metrics should have "_total" or "_timestamp_seconds" suffix
kubevirt_vmi_filesystem_capacity_bytes_total: non-counter metrics should not have "_total" suffix
kubevirt_vmi_memory_domain_bytes_total: non-counter metrics should not have "_total" suffix
kubevirt_vmi_memory_pgmajfault: counter metrics should have "_total" or "_timestamp_seconds" suffix
kubevirt_vmi_memory_pgminfault: counter metrics should have "_total" or "_timestamp_seconds" suffix
kubevirt_vmi_memory_swap_in_traffic_bytes_total: non-counter metrics should not have "_total" suffix
kubevirt_vmi_memory_swap_out_traffic_bytes_total: non-counter metrics should not have "_total" suffix
kubevirt_vmi_outdated_count: non-histogram and non-summary metrics should not have "_count" suffix
kubevirt_vmi_storage_flush_times_ms_total: metric names should not contain abbreviated units
kubevirt_vmi_storage_read_times_ms_total: metric names should not contain abbreviated units
kubevirt_vmi_storage_write_times_ms_total: metric names should not contain abbreviated units
kubevirt_vmi_vcpu_seconds: counter metrics should have "_total" or "_timestamp_seconds" suffix
kubevirt_vmi_vcpu_wait_seconds: counter metrics should have "_total" or "_timestamp_seconds" suffix
kubevirt_vmsnapshot_disks_restored_from_source_total: non-counter metrics should not have "_total" suffix
make: *** [Makefile:213: lint-metrics] Error 1
```

**What you expected to happen**:
Metrics named will be aligned with promlint linter and Prometheus best practices.

**How to reproduce it (as minimally and precisely as possible)**:
run make lint-metrics in kubevirt dir.

**Additional context**:
A bugzilla bug opened as well: [Bug 2196251 - Metrics names failed promlint linter](https://bugzilla.redhat.com/show_bug.cgi?id=2196251)

**Environment**:
- KubeVirt version (use `virtctl version`): 4.13
- Kubernetes version (use `kubectl version`): N/A
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by sradco**
Signed-off-by: Shirly Radco <sradco@redhat.com>

**What this PR does / why we need it**:
This PR updated metrics names to meet the metrics naming conventions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #9714

**Special notes for your reviewer**:

**Release note**:

```release-note
Deprecation notice for the metrics listed in the PR. Please update your systems to use the new metrics names.

|kubevirt_migrate_vmi_failed | kubevirt_vmi_migrations_failed|
|kubevirt_migrate_vmi_succeeded | kubevirt_vmi_migrations_succeeded|
|kubevirt_migrate_vmi_pending_count | kubevirt_vmi_migrations_pending|
|kubevirt_migrate_vmi_running_count | kubevirt_vmi_migrations_running|
|kubevirt_migrate_vmi_scheduling_count | kubevirt_vmi_migrations_scheduling|
|kubevirt_vmi_filesystem_capacity_bytes_total | kubevirt_vmi_filesystem_capacity_bytes|
|kubevirt_vmi_memory_domain_bytes_total | kubevirt_vmi_memory_domain_bytes|
|kubevirt_vmi_memory_pgmajfault | kubevirt_vmi_memory_pgmajfault_total|
|kubevirt_vmi_memory_pgminfault | kubevirt_vmi_memory_pgminfault_total|
|kubevirt_vmi_memory_swap_in_traffic_bytes_total | kubevirt_vmi_memory_swap_in_traffic_bytes|
|kubevirt_vmi_memory_swap_out_traffic_bytes_total | kubevirt_vmi_memory_swap_out_traffic_bytes|
|kubevirt_vmi_outdated_count | kubevirt_vmi_number_of_outdated|
|kubevirt_vmi_storage_flush_times_ms_total | kubevirt_vmi_storage_flush_times_seconds_total|
|kubevirt_vmi_storage_read_times_ms_total | kubevirt_vmi_storage_read_times_seconds_total|
|kubevirt_vmi_storage_write_times_ms_total | kubevirt_vmi_storage_write_times_seconds_total|
|kubevirt_vmi_vcpu_seconds | kubevirt_vmi_vcpu_seconds_total|
|kubevirt_vmi_vcpu_wait_seconds | kubevirt_vmi_vcpu_wait_seconds_total|
|kubevirt_vmsnapshot_disks_restored_from_source_total | kubevirt_vmsnapshot_disks_restored_from_source|
|kubevirt_allocatable_nodes_count | kubevirt_allocatable_nodes|
|kubevirt_kvm_available_nodes_count | kubevirt_nodes_with_kvm|
|kubevirt_virt_api_up_total | kubevirt_virt_api_up|
|kubevirt_virt_controller_ready_total | kubevirt_virt_controller_ready|
|kubevirt_virt_controller_ready | kubevirt_virt_controller_ready_status|
|kubevirt_virt_controller_up_total | kubevirt_virt_controller_up|
|kubevirt_virt_handler_up_total | kubevirt_virt_handler_up|
|kubevirt_virt_operator_leading_total | kubevirt_virt_operator_leading|
|kubevirt_virt_operator_leading | kubevirt_virt_operator_leading_status|
|kubevirt_virt_operator_ready_total | kubevirt_virt_operator_ready|
|kubevirt_virt_operator_ready | kubevirt_virt_operator_ready_status|
|kubevirt_virt_operator_up_total | kubevirt_virt_operator_up|

```

### CODE REVIEW

**Comment by sradco** [comment]
@machadovilaca @assafad @avlitman  Please review

**Comment by avlitman** [comment]
Hi @sradco I'm not sure how it happened since I rebased few times before running the linter but my fork was not up to date so I missed some new metrics that also failing the linter. 
The linter output after this changes is:
kubevirt_allocatable_nodes_count: non-histogram and non-summary metrics should not have "_count" suffix
kubevirt_kvm_available_nodes_count: non-histogram and non-summary metrics should not have "_count" suffix
kubevirt_virt_api_up_total: non-counter metrics should not have "_total" suffix
kubevirt_virt_controller_ready_total: non-counter metrics should not have "_total" suffix
kubevirt_virt_controller_up_total: non-counter metrics should not have "_total" suffix
kubevirt_virt_handler_up_total: non-counter metrics should not have "_total" suffix
kubevirt_virt_operator_leading_total: non-counter metrics should not have "_total" suffix
kubevirt_virt_operator_ready_total: non-counter metrics should not have "_total" suffix
kubevirt_virt_operator_up_total: non-counter metrics should not have "_total" suffix
kubevirt_vmi_phase_count: non-histogram and non-summary metrics should not have "_count" suffix

**Comment by sradco** [comment]
@avlitman please open the new metrics names that needs update in a separate issue, for an easier review.
 Thank you.

**Comment by avlitman** [comment]
> kubevirt_allocatable_nodes_count: non-histogram and non-summary metrics should not have "_count" suffix
> kubevirt_kvm_available_nodes_count: non-histogram and non-summary metrics should not have "_count" suffix
> kubevirt_virt_api_up_total: non-counter metrics should not have "_total" suffix
> kubevirt_virt_controller_ready_total: non-counter metrics should not have "_total" suffix
> kubevirt_virt_controller_up_total: non-counter metrics should not have "_total" suffix
> kubevirt_virt_handler_up_total: non-counter metrics should not have "_total" suffix
> kubevirt_virt_operator_leading_total: non-counter metrics should not have "_total" suffix
> kubevirt_virt_operator_ready_total: non-counter metrics should not have "_total" suffix
> kubevirt_virt_operator_up_total: non-counter metrics should not have "_total" suffix
> kubevirt_vmi_phase_count: non-histogram and non-summary metrics should not have "_count" suffix

New issue: https://github.com/kubevirt/kubevirt/issues/9836
New bug: https://bugzilla.redhat.com/show_bug.cgi?id=2211671

**Comment by avlitman** [comment]
> @avlitman please open the new metrics names that needs update in a separate issue, for an easier review. Thank you.

New issue: https://github.com/kubevirt/kubevirt/issues/9836
New bug: https://bugzilla.redhat.com/show_bug.cgi?id=2211671

File: pkg/virt-operator/resource/generate/components/prometheus.go
Comment (machadovilaca) [inline pkg/virt-operator/resource/generate/components/prometheus.go:882]: maybe can we move these to a separate function, maybe `GetDeprecatedRecordingRules`, to have a clear distinction for the deprecated items?

maybe even somehow include the last version they were not deprecated -> `GetDeprecatedRecordingRulesAfterRelease59`

that would make it easier to track the metrics to completely remove after 2 minor releases or something, and it would be very simple to remove them

Comment (sradco) [reply pkg/virt-operator/resource/generate/components/prometheus.go]: I updated the PR so that the new metric names will be the recording rules and in 2 versions we would replace the metric names and remove the recording rules.
I did this since the help text is not visible for recording rule, only for metrics.

**Comment by sradco** [comment]
/retest-required

pull-kubevirt-code-lint  isn't failing on something that is related to this PR.

**Comment by sradco** [comment]
@machadovilaca @vladikr @lyarwood Please review.

**Comment by lyarwood** [comment]
I don't really have any comments on the new names but I've got a few nits below regarding the PR description and release note:

> **What this PR does / why we need it**: This PR updated metrics names to meet the metrics naming conventions. The old metrics names will still be available in order to allow the users to update their dashboards, but will be deprecated in a few months.

I think you mean `removed in a future release` instead of `deprecated` right?

> **Release note**:
> 
> ```
> Deprecation notice for the metrics listed in the PR. They will be removed in the next releases. Please update your systems to use the new metrics names.
> ```

Can we list the deprecated metrics in the docs somewhere and link to that here? Referencing the PR isn't helpful IMHO.

Also as above I'd change this to  `They will be removed in a future release`.

**Comment by sradco** [comment]
@machadovilaca @vladikr @lyarwood Please review.

**Comment by lyarwood** [comment]
@sradco Apologies but I'm slightly confused, are we deprecating or just straight up renaming these metrics? I can't see any evidence of the original metrics in the tree with your PR applied?

```
$ gh pr checkout 9821
$ git grep kubevirt_migrate_vmi_failed
$ 
```

**Comment by sradco** [comment]
We are only updating the metrics names.

**Comment by lyarwood** [comment]
> We are only updating the metrics names.

Okay the release note confused me, if you can remove the suggestion of deprecation from that then this `lgtm`.

**Comment by sradco** [comment]
@machadovilaca @enp0s3 @iholder101 Please review. I don't understand why these tests are failing.

**Review by avlitman — APPROVED** [review]

**Comment by iholder101** [comment]
@sradco The tests failed multiple times with same errors. I believe retesting further won't help.

**Comment by enp0s3** [comment]
/hold
/cc
The problem with the monitoring lane. The `NoReadyVirtOperator` alert is still there although tests expect it to be gone.
The problem is that we don't have k8s-reporter artifacts in that case because no running VMs are involved.

I will rebase this PR and test it locally. However I will do it later on today due to some backlog.

**Comment by sradco** [comment]
@enp0s3 @lyarwood @iholder101 please review. I don't think the issue with the tests is related to this PR. Thank you.

**Comment by enp0s3** [comment]
Indeed the [storage job](https://prow.ci.kubevirt.io/job-history/gs/kubevirt-prow/pr-logs/directory/pull-kubevirt-e2e-k8s-1.26-sig-storage) is flaky.

**Review by enp0s3 — APPROVED** [review]
/approve

**Comment by sradco** [comment]
@enp0s3 @avlitman, can you please cherrypick this PR to 4.14 after it is merged?

**Comment by enp0s3** [comment]
@akalenyu @awels @ShellyKa13 Hey! can you please keep me honest about the sig-storage being flaky on this PR? I saw it fail 4 times in a row, each time on a different test.

**Comment by awels** [comment]
Not aware of any particular flaky test in the storage lane.

**Comment by enp0s3** [comment]
@avlitman Lets wait with the retest, I will either override it or there is a real issue, one way or another we need assistance from the storage folks.

**Comment by sradco** [comment]
@avlitman @enp0s3 please backport this PR to 4.14

**Comment by avlitman** [comment]
@enp0s3 Should I cherry-pick for both 0.59 and 1.0 like in this latest Joao's re-naming metrics pr: https://github.com/kubevirt/kubevirt/pull/10138?

**Comment by enp0s3** [comment]
@avlitman @assafad @machadovilaca Hi, please create a manual backport PR.

**Pull request merged**
